File: lib/models/trending_entry_destiny_activity.dart
import 'destiny_public_activity_status.dart';
class TrendingEntryDestinyActivity{
	int activityHash;
	DestinyPublicActivityStatus status;
	TrendingEntryDestinyActivity(
		this.activityHash,
		this.status,
	);

	static TrendingEntryDestinyActivity fromMap(Map<String, dynamic> data){
		if(data == null) {
			return null;
		};
		return new TrendingEntryDestinyActivity(
				data['activityHash'],
				data['status'] != null ? DestinyPublicActivityStatus.fromMap(data['status']) : null,
		);
	}

	static List<TrendingEntryDestinyActivity> fromList(List<dynamic> data){
		if(data == null) {
			return null;
		};
		List<TrendingEntryDestinyActivity> list = new List();
    data.forEach((item) {
      list.add(TrendingEntryDestinyActivity.fromMap(item));
    });
    return list;
	}

	Map<String, dynamic> toMap(){
		Map<String, dynamic> data = new Map();
			data['activityHash'] = this.activityHash;
			data['status'] = this.status.toMap();
		return data;
	}
}

File: BungieNetApi/Models/TrendingEntryDestinyActivity.cs
using System.Runtime.Serialization;

namespace GhostSharp.BungieNetApi.Model
{
	[DataContract]
	public class TrendingEntryDestinyActivity{

		[DataMember(Name="activityHash", EmitDefaultValue=false)]
		public uint ActivityHash { get; set; }

		/// <summary>
		/// Represents the public-facing status of an activity: any data about what is currently active in the Activity, regardless of an individual character's progress in it.
		/// </summary>
		[DataMember(Name="status", EmitDefaultValue=false)]
		public DestinyPublicActivityStatus Status { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as TrendingEntryDestinyActivity);
        }

		public bool Equals(TrendingEntryDestinyActivity input)
		{
			if (input == null) return false;

			return
				(
                    ActivityHash == input.ActivityHash ||
                    (ActivityHash != null && ActivityHash.Equals(input.ActivityHash))
                ) &&
				(
                    Status == input.Status ||
                    (Status != null && Status.Equals(input.Status))
                ) ;
		}
	}
}

